test/net/sourceforge/kolmafia/preferences/PreferencesTest.java

```diff
@@ -651,34 +710,46 @@ public void exerciseResetDots() {
   }
 
   @Nested
-  class SaveSettingsOnSet {
+  class SaveTogglePreferencesTest {
+    private final String USER_NAME = "PreferencesTestAlsoFakeUser".toLowerCase();
+
+    @BeforeEach
+    public void initializeCharPreferences() {
+      KoLCharacter.reset(USER_NAME);
+    }
+
+    @AfterEach
+    public void resetCharAndPreferences() {
+      File userFile = new File("settings/" + USER_NAME + "_prefs.txt");
+      verboseDelete(userFile);
+      File backupFile = new File("settings/" + USER_NAME + "_prefs.bak");
+      verboseDelete(backupFile);
+    }
+
     @Test
     public void savesSettingsIfOn() throws IOException {
       var cleanups =
           new Cleanups(withSavePreferencesToFile(), withProperty("saveSettingsOnSet", true));
       try (cleanups) {
-        File userFile =
-            new File("settings/" + KoLCharacter.getUserName().toLowerCase() + "_prefs.txt");
-        String contents =
-            new String(
-                DataUtilities.getInputStream(userFile).readAllBytes(), StandardCharsets.UTF_8);
+        File userFile = new File("settings/" + USER_NAME + "_prefs.txt");
+        InputStream inputStream = DataUtilities.getInputStream(userFile);
+        String contents = new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+        inputStream.close();
         assertThat(contents, not(containsString("\nxyz=abc\n")));
-
-        try (var cleanups2 = withProperty("xyz", "abc")) {
-          contents =
-              new String(
-                  DataUtilities.getInputStream(userFile).readAllBytes(), StandardCharsets.UTF_8);
-          assertThat(contents, containsString("\nxyz=abc\n"));
-        }
+        Preferences.setString("xyz", "abc");
```

Comment:
cleanups is preferred

Reply:
I might disagree?  The test allows preferences to be written to a file when they are set.  The test reads the file, verifies a preference is not set there, sets a preference, reads the file and verifies that the new preference was written to the file.

Maybe I don't understand the sequencing of cleanups but I didn't use it because I could not guarantee that it would not be in the first read.

Reply:
you could put cleanups on this line, not in the first cleanups.

But this is also fine because this is what you're testing, and "xyz" doesn't need to be reset as cleanups would do.